src/components/cesium_map/cesiumh3.js

```diff
@@ -120,19 +122,30 @@ export class H3GridManager {
             this.global_grid = new H3Grid();
         }
         this.global_grid.rect_str = rstr;
+        // delete old grid
+        let toRemove = cview.dataSources.getByName(this.old_grid_rstr)[0];
+        cview.dataSources.remove(toRemove, true);
         const _this = this;
         this.global_grid.load(rstr).then((ds) => {
-            // delete existing grid
             try {
-                cview.dataSources.remove(_this.current_grid.data, true);
                 cview.dataSources.add(ds);
                 _this.current_grid = this.global_grid;
                 _this.current_rect = rect;
-                _this.old_grid_data = this.global_grid.data; 
+                _this.old_grid_rstr = ds.name; // update 
             } catch (err) {
                 console.log(err);
             }
-        });
+            console.log("then loop: ",cview.dataSources)
+        }).then(()=>{
+            // remove all the sources that are not the current grid layer 
```

Comment:
Ensure that only one grid data source is rendered by doing a clean up phase. This will prevent duplicate grid data sources.